CONTRIBUTING.md

```diff
@@ -45,22 +45,132 @@ To add a new plugin you need to do the following.
+Such plugins are usually configured via vim globals, but often have no configurable options at all.
+
+| Parameter                | Description                                                                 | Required | Default Value |
+|--------------------------|-----------------------------------------------------------------------------|----------|---------------|
+| **name**                 | The name of the Vim plugin.                                                 | Yes      | N/A           |
+| **url**                  | The URL of the plugin repository.                                           | Yes      | N/A           |
+| **colorscheme**          | The name of the colorscheme.                                                | No       | `name` parameter |
+| **deprecateExtraConfig** | Flag to deprecate extra configuration.                                      | No       | `false`       |
+| **description**          | A description of the plugin. Can also be used for non-normative documentation, warnings, tips and tricks.                                                | No       | `null`        |
+| **extraConfig**          | Extra configuration for the plugin.                                         | No       | `cfg: {}`     |
+| **extraOptions**         | Extra options for the plugin.                                               | No       | `{}`          |
+| **extraPackages**        | Extra packages to include.                                                  | No       | `[]`          |
+| **extraPlugins**         | Extra plugins to include.                                                   | No       | `[]`          |
+| **globalPrefix**         | Global prefix for the settings.                                             | No       | `""`          |
+| **imports**              | A list of imports for the plugin.                                           | No       | `[]`          |
+| **isColorscheme**        | Flag to indicate if the plugin is a colorscheme.                            | No       | `false`       |
+| **maintainers**          | The maintainers of the plugin.                                              | No      | `[]` |
+| **optionsRenamedToSettings** | List of options renamed to settings.                                    | No       | `[]`          |
+| **originalName**         | The original name of the plugin, usually the plugin's github repo name. E.g. `"foo-bar.nvim"`.                                            | No       | `name` parameter |
+| **package**              | The package for the plugin.                                                 | No       | `name` parameter |
+| **settingsExample**      | Example settings for the plugin.                                            | No       | `null`        |
+| **settingsOptions**      | Settings options for the plugin.                                            | No       | `{}`          |
```

Comment:
```suggestion
| **settingsOptions**      | Options representing the plugin's settings. This is optional because `settings` is a "freeform" option. See [Declaring plugin options](#declaring-plugin-options).    | No       | `{}`          |
```

Should probably match the equivalent in `mkNeovimPlugin`